server/src/main/java/org/elasticsearch/bootstrap/Bootstrap.java

```diff
@@ -234,14 +236,16 @@ static SecureSettings loadSecureSettings(Environment initialEnv) throws Bootstra
             throw new BootstrapException(e);
         }
 
-        try {
+        try (SecureString password = keystore != null && keystore.hasPassword()
+                ? new SecureString(Terminal.DEFAULT.readSecret("Elasticsearch keystore password? "))
```

Comment:
I think we should do this prompting from bin/elasticsearch, and always read from stdin here.